tests/test_other.py

```diff
@@ -12226,3 +12226,54 @@ def test_rust_gxx_personality_v0(self):
         }
       }
     ''', assert_returncode=NON_ZERO)
+
+  def test_safari_bigint64array_shim(self):
+    bigint64array = Path(path_from_root("src/polyfill/bigint64array.js")).read_text()
+    test_code = read_file(test_file('test_safari_bigint64array_shim.js'))
+    bigint_list = [
+      0,
+      1,
+      -1,
+      5,
+      (1 << 64),
+      (1 << 64) - 1,
+      (1 << 64) + 1,
+      (1 << 63),
+      (1 << 63) - 1,
+      (1 << 63) + 1,
+    ]
+    bigint_list_strs = [str(x) for x in bigint_list]
+
+    bigint_list_unsigned = [x % (1 << 64) for x in bigint_list]
+    bigint_list_signed = [
+      x if x < 0 else (x % (1 << 64)) - 2 * (x & (1 << 63)) for x in bigint_list
+    ]
+    bigint_list_unsigned_n = [f"{x}n" for x in bigint_list_unsigned]
+    bigint_list_signed_n = [f"{x}n" for x in bigint_list_signed]
+
+    bigint64array = "".join(bigint64array.splitlines(keepends=True)[3:])
```

Comment:
I was going to ask how you made this work without running the pre-processor over the JS file.  But it looks like you just trip the first three lines.   Which seem a little fragile.

Maybe better run a integrated test?   Use `emcc` with the test code as a `--post-js` files?

(If you do keep this line then I think maybe `"\n".join(...)` without the `keepends=True` would be more pythonic)

Reply:
Well I want to directly compare the behavior of the native `BigInt64Array` with my version, which I think would be hard to do with the integrated test. I guess I could delete `BigInt64Array` in `--extern-pre-js` or something? Deleting the `#ifdef` seems fine to me for now.